core/src/blocktree.rs

```diff
@@ -977,6 +1040,53 @@ impl Blocktree {
         }
     }
 
+    /// Checks if this data blob is a duplicate. If  it's not, then inserts it.
```

Comment:
A more accurate description: "Checks to see if the data blob passes integrity checks for insertion"